generator/generator.py

```diff
@@ -7,9 +7,13 @@
 
 
 class Generator(ABC):
-    def __init__(self):
+    def __init__(self, name=None):
```

Comment:
can we just change this parameter to `instance_name`?

Reply:
We could, but I was thinking that the use of this in user code is when instancing the `Generator` object. So, say we had `class MyGenerator(Generator):`, then instancing it with a name would be: `dummy = MyGenerator(name="dummy")`. I thought that including the `instance_` prefix in the user facing code would be a bit too verbose, but internally it would be useful to be explicit about the fact that it's the instance name (versus the definition name).

Reply:
That makes sense. Just `name` is good.